clang/include/clang/Basic/TargetInfo.h

```diff
@@ -1211,6 +1212,14 @@ class TargetInfo : public TransferrableTargetInfo,
       TiedOperand = N;
       // Don't copy Name or constraint string.
     }
+
+    // CC range can be set by targets supporting flag output operand.
+    void setFlagOutputCCUpperBound(unsigned CCBound) {
```

Comment:
I think it might actually be preferable to implement more generic infrastructure - there nothing in here that necessarily ties the bounds to a flag output operand.  We might as well write the infrastructure to allow setting guaranteed bounds (probably lower and higher?) for *any* output operand.  Targets might chose to use this for other output operands where a constraint guarantees the output value must fall within a certain range.  That would also be more symmetrical to the use of bounds for input operands.